Readme-developer.md

```diff
@@ -47,6 +49,28 @@ https://documentation.dnanexus.com/downloads
     make
     ```
 
+# Building inside docker 
+
+To avoid lengthy installation of dependencies on OS X and simultaneous installations of development versions of `dx-toolkit`, you can build your dev-version of `dx-toolkit` in the docker container. 
+
+1. Start `python:3.9-bullseye` in the interactive mode, mounting the repo you are working on (`<local_path_to_repo>/dx-toolkit`):
+
+    ```
+    # from root folder of dx-tollkit
+    docker run -v `pwd`:/dx-toolkit -w /dx-toolkit -it --rm --entrypoint=/bin/bash python:3.9-bullseye
+    ```
+2. From the interactive shell install `dx-toolkit`:
+
+    ```
+    python -m pip install src/python/ --upgrade
+    ```
+3. Log in and use the container while developing. To rebuild, just save the work and run the step 2 again. 
+
+Also, you can isntall the the `dx-toolkit` from remote like so: 
+    ```
+    python3 -m pip install --upgrade 'git+https://github.com/dnanexus/dx-toolkit.git@master#egg=dxpy&subdirectory=src/python'
```

Comment:
This might be 2a (or 2b), so user immediately knows there are more options?

Reply:
Sure.